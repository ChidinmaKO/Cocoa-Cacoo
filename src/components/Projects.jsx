import React, { Component } from 'react';
import { Grid, GridCol } from 'griz';
import axios from 'axios';
import Header from './Header';
import List from './List';
import Button from './Button';
import { FloatRight, LineDivide, Avatar } from './Util';


const host = process.env.HOST;
const apikey = process.env.CACOO_API;

class Projects extends Component {
  constructor() {
    super();
    this.state = {
      diagramList: []
    };
    this.deleteDiagram = this.deleteDiagram.bind(this);
  }

  /* eslint-disable no-console */
  componentDidMount() {
    axios.get(`${host}/diagrams.json?apiKey=${apikey}`)
      .then(response => {
        const { data: { result } } = response;
        this.setState({
          diagramList: result
        });
        // console.log(result);
      })
      .catch(error => {
        console.log(error);
      });
  }

  deleteDiagram(diagram) {
    const { diagramList } = this.state;
    this.setState({
      diagramList: diagramList.filter(item => item.diagramId !== diagram)
    });
    axios.get(`${host}/diagrams/${diagram}/delete.json?apiKey=${apikey}`)
      .then(response => {
        console.log(response);
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    const { diagramList } = this.state;
    return (
      <React.Fragment>
        <Header>
          <li>Manage Projects</li>
          <li>me</li>
          <li>Logout</li>
        </Header>
        <Grid responsiveSm gutterWidth="60">
          <GridCol column="70">
            <Grid gutterless>
              <GridCol>
                <h4>My Projects</h4>
              </GridCol>
              {/* <GridCol>
                <FloatRight>
                  <Button type="button">Create</Button>
                </FloatRight>
              </GridCol> */}
            </Grid>
            <List>
              {diagramList.map(diagram => (
                <li key={diagram.diagramId}>
                  <details>
                    <summary>
                      <LineDivide spaced>
                        <h4>{ diagram.title }</h4>
                        <button onClick={() => this.deleteDiagram(diagram.diagramId)} type="button">Delete</button>
                      </LineDivide>
                    </summary>
                    <div>
                      <Grid responsiveMd>
                        <GridCol column="32">
                          <img src={diagram.imageUrl} alt="project" />
                        </GridCol>
                        <GridCol>
                          <h4>{ diagram.title }</h4>
                          <h6>
                            Folder Name:
                            { diagram.folderName }
                          </h6>
                          <time>{ diagram.created }</time>
                          <p>
                            Lorem ipsum dolor sit amet,
                            consectetur adipiscing elit,
                            ut labore et dolore magna aliqua. Ut enim ad minim veniam,
                            quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
                            commodo consequat.
                          </p>
                        </GridCol>
                      </Grid>
                    </div>
                  </details>
                </li>
              ))}
            </List>
          </GridCol>
          <GridCol>
            <Grid gutterless>
              <GridCol>
                <h4>User Information</h4>
              </GridCol>
            </Grid>
            <List padded>
              <li>
                <LineDivide>
                  <Avatar src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32" alt="img" />
                  <h5>
                    Name: Chidinma K.
                  </h5>
                </LineDivide>
              </li>
            </List>
          </GridCol>
        </Grid>
      </React.Fragment>
    );
  }
}

export default Projects;
