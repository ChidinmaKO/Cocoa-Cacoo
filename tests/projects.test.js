import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Projects from '../src/components/Projects';

Enzyme.configure({ adapter: new Adapter() });

/* eslint-disable  */
describe('Projects component', () => {
  test('renders', () => {
    const wrapper = shallow(<Projects />);

    expect(wrapper.exists()).toBe(true);
  });

  test('matches the snapshot', () => {
    const wrapper = shallow(<Projects />);

    expect(wrapper).toMatchSnapshot();
  });

  test('there are projects listed', () => {
    const wrapper = shallow(<Projects />);

    expect(wrapper.find('li')).not.toHaveLength(0);
  });

  test('the projects listed have a title', () => {
    const wrapper = shallow(<Projects/>);

    expect(wrapper.find('h4').exists()).toBe(true);
  });

  test('there is a user information', () => {
    const wrapper = shallow(<Projects/>);

    expect(wrapper.find('h5').text()).toBe('Name: Chidinma K.');
  });

  test('there is a delete button', () => {
    const wrapper = shallow(<Projects deleteDiagram={() => {}} />);

    const button = wrapper.find('button');

    expect(button.exists()).toBeDefined;
  });
});
