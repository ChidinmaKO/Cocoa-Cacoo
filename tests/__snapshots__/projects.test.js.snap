// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navigation>
          <li>
            Manage Projects
          </li>
          <li>
            me
          </li>
          <li>
            Logout
          </li>
        </Navigation>,
        <styled.div
          gutterWidth="60"
          responsiveSm={true}
        >
          <styled.div
            column="70"
          >
            <styled.div
              gutterless={true}
            >
              <styled.div>
                <h4>
                  My Projects
                </h4>
              </styled.div>
            </styled.div>
            <ForwardRef />
          </styled.div>
          <styled.div>
            <styled.div
              gutterless={true}
            >
              <styled.div>
                <h4>
                  User Information
                </h4>
              </styled.div>
            </styled.div>
            <ForwardRef
              padded={true}
            >
              <li>
                <ForwardRef>
                  <ForwardRef
                    alt="img"
                    src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                  />
                  <h5>
                    Name: Chidinma K.
                  </h5>
                </ForwardRef>
              </li>
            </ForwardRef>
          </styled.div>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <li>
              Manage Projects
            </li>,
            <li>
              me
            </li>,
            <li>
              Logout
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Manage Projects",
            },
            "ref": null,
            "rendered": "Manage Projects",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "me",
            },
            "ref": null,
            "rendered": "me",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
            },
            "ref": null,
            "rendered": "Logout",
            "type": "li",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              column="70"
            >
              <styled.div
                gutterless={true}
              >
                <styled.div>
                  <h4>
                    My Projects
                  </h4>
                </styled.div>
              </styled.div>
              <ForwardRef />
            </styled.div>,
            <styled.div>
              <styled.div
                gutterless={true}
              >
                <styled.div>
                  <h4>
                    User Information
                  </h4>
                </styled.div>
              </styled.div>
              <ForwardRef
                padded={true}
              >
                <li>
                  <ForwardRef>
                    <ForwardRef
                      alt="img"
                      src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                    />
                    <h5>
                      Name: Chidinma K.
                    </h5>
                  </ForwardRef>
                </li>
              </ForwardRef>
            </styled.div>,
          ],
          "gutterWidth": "60",
          "responsiveSm": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div
                  gutterless={true}
                >
                  <styled.div>
                    <h4>
                      My Projects
                    </h4>
                  </styled.div>
                </styled.div>,
                <ForwardRef />,
              ],
              "column": "70",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <styled.div>
                    <h4>
                      My Projects
                    </h4>
                  </styled.div>,
                  "gutterless": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h4>
                      My Projects
                    </h4>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "My Projects",
                    },
                    "ref": null,
                    "rendered": "My Projects",
                    "type": "h4",
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "rules": Array [
                      "
  list-style: none;
  margin: 10px 0 0;
  padding: 0;
  li {
    background: #ffffff;
    border: solid thin #bbb;
    border-bottom: 0;
    ",
                      [Function],
                      "
    &:last-of-type{
      border-bottom: solid thin #bbb;
    }
  }
  summary {
    padding: 10px 20px;
    &::-webkit-details-marker{
      display: none;
    }
    &:focus{
      outline: solid thin #aaa;
    }
  }
  details {
    > div {
      border-top: solid thin #bbb;
      padding: 10px;
      background: #f5f5f5;
      h4 {
        margin-top: 0;
      }
    }
  }
",
                    ],
                  },
                  "displayName": "styled.ul",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "ul",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div
                  gutterless={true}
                >
                  <styled.div>
                    <h4>
                      User Information
                    </h4>
                  </styled.div>
                </styled.div>,
                <ForwardRef
                  padded={true}
                >
                  <li>
                    <ForwardRef>
                      <ForwardRef
                        alt="img"
                        src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                      />
                      <h5>
                        Name: Chidinma K.
                      </h5>
                    </ForwardRef>
                  </li>
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <styled.div>
                    <h4>
                      User Information
                    </h4>
                  </styled.div>,
                  "gutterless": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h4>
                      User Information
                    </h4>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "User Information",
                    },
                    "ref": null,
                    "rendered": "User Information",
                    "type": "h4",
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <li>
                    <ForwardRef>
                      <ForwardRef
                        alt="img"
                        src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                      />
                      <h5>
                        Name: Chidinma K.
                      </h5>
                    </ForwardRef>
                  </li>,
                  "padded": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <ForwardRef>
                      <ForwardRef
                        alt="img"
                        src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                      />
                      <h5>
                        Name: Chidinma K.
                      </h5>
                    </ForwardRef>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <ForwardRef
                          alt="img"
                          src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                        />,
                        <h5>
                          Name: Chidinma K.
                        </h5>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "alt": "img",
                          "src": "https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bZQynM",
                            "isStatic": true,
                            "rules": Array [
                              "
  border-radius: 50%;
  margin-right: 20px;
  width: 50px;
  height: 50px;
",
                            ],
                          },
                          "displayName": "styled.img",
                          "render": [Function],
                          "styledComponentId": "sc-bZQynM",
                          "target": "img",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        },
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Name: Chidinma K.",
                        },
                        "ref": null,
                        "rendered": "Name: Chidinma K.",
                        "type": "h5",
                      },
                    ],
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "rules": Array [
                          "
  display: flex;
  ",
                          [Function],
                          "
  align-items: center;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    },
                  },
                  "type": "li",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "rules": Array [
                      "
  list-style: none;
  margin: 10px 0 0;
  padding: 0;
  li {
    background: #ffffff;
    border: solid thin #bbb;
    border-bottom: 0;
    ",
                      [Function],
                      "
    &:last-of-type{
      border-bottom: solid thin #bbb;
    }
  }
  summary {
    padding: 10px 20px;
    &::-webkit-details-marker{
      display: none;
    }
    &:focus{
      outline: solid thin #aaa;
    }
  }
  details {
    > div {
      border-top: solid thin #bbb;
      padding: 10px;
      background: #f5f5f5;
      h4 {
        margin-top: 0;
      }
    }
  }
",
                    ],
                  },
                  "displayName": "styled.ul",
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "ul",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navigation>
            <li>
              Manage Projects
            </li>
            <li>
              me
            </li>
            <li>
              Logout
            </li>
          </Navigation>,
          <styled.div
            gutterWidth="60"
            responsiveSm={true}
          >
            <styled.div
              column="70"
            >
              <styled.div
                gutterless={true}
              >
                <styled.div>
                  <h4>
                    My Projects
                  </h4>
                </styled.div>
              </styled.div>
              <ForwardRef />
            </styled.div>
            <styled.div>
              <styled.div
                gutterless={true}
              >
                <styled.div>
                  <h4>
                    User Information
                  </h4>
                </styled.div>
              </styled.div>
              <ForwardRef
                padded={true}
              >
                <li>
                  <ForwardRef>
                    <ForwardRef
                      alt="img"
                      src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                    />
                    <h5>
                      Name: Chidinma K.
                    </h5>
                  </ForwardRef>
                </li>
              </ForwardRef>
            </styled.div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <li>
                Manage Projects
              </li>,
              <li>
                me
              </li>,
              <li>
                Logout
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Manage Projects",
              },
              "ref": null,
              "rendered": "Manage Projects",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "me",
              },
              "ref": null,
              "rendered": "me",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
              },
              "ref": null,
              "rendered": "Logout",
              "type": "li",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                column="70"
              >
                <styled.div
                  gutterless={true}
                >
                  <styled.div>
                    <h4>
                      My Projects
                    </h4>
                  </styled.div>
                </styled.div>
                <ForwardRef />
              </styled.div>,
              <styled.div>
                <styled.div
                  gutterless={true}
                >
                  <styled.div>
                    <h4>
                      User Information
                    </h4>
                  </styled.div>
                </styled.div>
                <ForwardRef
                  padded={true}
                >
                  <li>
                    <ForwardRef>
                      <ForwardRef
                        alt="img"
                        src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                      />
                      <h5>
                        Name: Chidinma K.
                      </h5>
                    </ForwardRef>
                  </li>
                </ForwardRef>
              </styled.div>,
            ],
            "gutterWidth": "60",
            "responsiveSm": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div
                    gutterless={true}
                  >
                    <styled.div>
                      <h4>
                        My Projects
                      </h4>
                    </styled.div>
                  </styled.div>,
                  <ForwardRef />,
                ],
                "column": "70",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <styled.div>
                      <h4>
                        My Projects
                      </h4>
                    </styled.div>,
                    "gutterless": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <h4>
                        My Projects
                      </h4>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "My Projects",
                      },
                      "ref": null,
                      "rendered": "My Projects",
                      "type": "h4",
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "rules": Array [
                        "
  list-style: none;
  margin: 10px 0 0;
  padding: 0;
  li {
    background: #ffffff;
    border: solid thin #bbb;
    border-bottom: 0;
    ",
                        [Function],
                        "
    &:last-of-type{
      border-bottom: solid thin #bbb;
    }
  }
  summary {
    padding: 10px 20px;
    &::-webkit-details-marker{
      display: none;
    }
    &:focus{
      outline: solid thin #aaa;
    }
  }
  details {
    > div {
      border-top: solid thin #bbb;
      padding: 10px;
      background: #f5f5f5;
      h4 {
        margin-top: 0;
      }
    }
  }
",
                      ],
                    },
                    "displayName": "styled.ul",
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "ul",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.div
                    gutterless={true}
                  >
                    <styled.div>
                      <h4>
                        User Information
                      </h4>
                    </styled.div>
                  </styled.div>,
                  <ForwardRef
                    padded={true}
                  >
                    <li>
                      <ForwardRef>
                        <ForwardRef
                          alt="img"
                          src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                        />
                        <h5>
                          Name: Chidinma K.
                        </h5>
                      </ForwardRef>
                    </li>
                  </ForwardRef>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <styled.div>
                      <h4>
                        User Information
                      </h4>
                    </styled.div>,
                    "gutterless": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <h4>
                        User Information
                      </h4>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "User Information",
                      },
                      "ref": null,
                      "rendered": "User Information",
                      "type": "h4",
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <li>
                      <ForwardRef>
                        <ForwardRef
                          alt="img"
                          src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                        />
                        <h5>
                          Name: Chidinma K.
                        </h5>
                      </ForwardRef>
                    </li>,
                    "padded": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <ForwardRef>
                        <ForwardRef
                          alt="img"
                          src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                        />
                        <h5>
                          Name: Chidinma K.
                        </h5>
                      </ForwardRef>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <ForwardRef
                            alt="img"
                            src="https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32"
                          />,
                          <h5>
                            Name: Chidinma K.
                          </h5>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "alt": "img",
                            "src": "https://cacoo.com/account/fSSuxzncJY40cCJd/image/32x32",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bZQynM",
                              "isStatic": true,
                              "rules": Array [
                                "
  border-radius: 50%;
  margin-right: 20px;
  width: 50px;
  height: 50px;
",
                              ],
                            },
                            "displayName": "styled.img",
                            "render": [Function],
                            "styledComponentId": "sc-bZQynM",
                            "target": "img",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          },
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Name: Chidinma K.",
                          },
                          "ref": null,
                          "rendered": "Name: Chidinma K.",
                          "type": "h5",
                        },
                      ],
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-EHOje",
                          "isStatic": false,
                          "rules": Array [
                            "
  display: flex;
  ",
                            [Function],
                            "
  align-items: center;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "render": [Function],
                        "styledComponentId": "sc-EHOje",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      },
                    },
                    "type": "li",
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "rules": Array [
                        "
  list-style: none;
  margin: 10px 0 0;
  padding: 0;
  li {
    background: #ffffff;
    border: solid thin #bbb;
    border-bottom: 0;
    ",
                        [Function],
                        "
    &:last-of-type{
      border-bottom: solid thin #bbb;
    }
  }
  summary {
    padding: 10px 20px;
    &::-webkit-details-marker{
      display: none;
    }
    &:focus{
      outline: solid thin #aaa;
    }
  }
  details {
    > div {
      border-top: solid thin #bbb;
      padding: 10px;
      background: #f5f5f5;
      h4 {
        margin-top: 0;
      }
    }
  }
",
                      ],
                    },
                    "displayName": "styled.ul",
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "ul",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
